tests/SchemaVersionTest.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Laravel\Tests;
+
+use MongoDB\Laravel\Tests\Models\SchemaVersion;
+use MongoDB\Laravel\Tests\Models\User;
+
+class SchemaVersionTest extends TestCase
+{
+    public function tearDown(): void
+    {
+        SchemaVersion::truncate();
+    }
+
+    public function testWithBasicDocument()
+    {
+        $document = new SchemaVersion(['name' => 'Luc']);
+        $this->assertEmpty($document->getSchemaVersion());
+        $document->save();
+
+        $this->assertEquals(1, $document->getSchemaVersion());
+        $this->assertNull($document->age);
+
+        $document->currentSchemaVersion = 2;
```

Comment:
You should not update this value directly. That's the responsability of `migrateSchema`.
```suggestion
```